src/machine.rs

```diff
@@ -482,11 +481,31 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
         let rng = StdRng::seed_from_u64(config.seed.unwrap_or(0));
         let borrow_tracker = config.borrow_tracker.map(|bt| bt.instanciate_global_state(config));
         let data_race = config.data_race_detector.then(|| data_race::GlobalState::new(config));
+        let page_size = if let Some(page_size) = config.page_size {
+            page_size
+        } else {
+            let target = &layout_cx.tcx.sess.target;
+            match target.arch.as_ref() {
+                "wasm32" | "wasm64" => 64 * 1024, // https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances
+                "aarch64" =>
+                    if target.options.vendor.as_ref() == "apple" {
+                        // No "definitive" source, but see:
+                        // https://www.wwdcnotes.com/notes/wwdc20/10214/
+                        // https://github.com/ziglang/zig/issues/11308 etc.
```

Comment:
The zig issue is about linux on M1.

Reply:
M1 is weird because some features require 16k pages (IOMMU as far as I recall?) but this also applies retroactively to Apple's ARM chips going back a long way (see https://developer.apple.com/library/archive/documentation/Performance/Conceptual/ManagingMemory/Articles/AboutMemory.html detailing that ever since A7 and A8, aarch64-apple-* has defaulted to 16k pages) and hackintoshes on aarch64 aren't a thing that exists right now AFAICT

Reply:
> M1 is weird because some features require 16k pages (IOMMU as far as I recall?)

Yeah and Linux can't deal with mixed page sizes, unlike macOS which is just fine with 16k for IOMMU and 4k for Rosetta 2.

Reply:
The issue I think still highlights that aarch64-apple prefers 16k pages by default